src/meta/src/hummock/compaction/picker/space_reclaim_compaction_picker.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_pb::hummock::hummock_version::Levels;
+use risingwave_pb::hummock::InputLevel;
+
+use crate::hummock::compaction::{CompactionInput, CompactionPicker, LocalPickerStatistic};
+use crate::hummock::level_handler::LevelHandler;
+
+pub struct SpaceReclaimCompactionPicker {
+    max_space_reclaim_file_count: usize,
+}
+
+impl SpaceReclaimCompactionPicker {
+    pub fn new(max_space_reclaim_file_count: usize) -> Self {
+        Self {
+            max_space_reclaim_file_count,
+        }
+    }
+}
+
+impl CompactionPicker for SpaceReclaimCompactionPicker {
+    fn pick_compaction(
+        &self,
+        levels: &Levels,
+        level_handlers: &[LevelHandler],
+        _stats: &mut LocalPickerStatistic,
+    ) -> Option<CompactionInput> {
+        assert!(!levels.levels.is_empty());
+        let reclaimed_level = levels.levels.last().unwrap();
+        let mut select_input_ssts = vec![];
+        let level_handler = &level_handlers[reclaimed_level.level_idx as usize];
+
+        for sst in &reclaimed_level.table_infos {
+            if level_handler.is_pending_compact(&sst.id) {
+                continue;
+            }
+
+            select_input_ssts.push(sst.clone());
+            if select_input_ssts.len() == self.max_space_reclaim_file_count {
+                break;
+            }
+        }
+
+        if select_input_ssts.is_empty() {
+            return None;
+        }
+
+        Some(CompactionInput {
+            input_levels: vec![
+                InputLevel {
+                    level_idx: reclaimed_level.level_idx,
+                    level_type: reclaimed_level.level_type,
+                    table_infos: select_input_ssts,
+                },
+                InputLevel {
```

Comment:
Is this empty input level required?

Reply:
It makes no practical sense, but I want to be consistent with other `compact_task`